dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataMetadataConnector.java

```diff
@@ -33,36 +34,54 @@
 import com.google.edwmigration.dumper.plugin.ext.jdk.annotation.Description;
 import com.google.edwmigration.dumper.plugin.lib.dumper.spi.TeradataMetadataDumpFormat;
 import java.util.List;
-import java.util.Optional;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import java.util.OptionalLong;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang3.StringUtils;
 
 /** @author miguel */
 @AutoService({Connector.class, MetadataConnector.class})
 @Description("Dumps metadata from Teradata.")
 @RespectsArgumentDatabasePredicate
 @RespectsArgumentAssessment
-@RespectsInput(
-    order = 450,
-    arg = ConnectorArguments.OPT_TERADATA_MAX_TABLESIZEV_ROWS,
-    description = ConnectorArguments.TERADATA_MAX_TABLE_SIZE_V_ROWS_DESCRIPTION)
-@RespectsInput(
-    order = 450,
-    arg = ConnectorArguments.OPT_TERADATA_MAX_DATABASESV_USER_ROWS,
-    description = ConnectorArguments.TERADATA_MAX_DATABASES_V_USER_ROWS_DESCRIPTION)
-@RespectsInput(
-    order = 450,
-    arg = ConnectorArguments.OPT_TERADATA_MAX_DATABASESV_DB_ROWS,
-    description = ConnectorArguments.TERADATA_MAX_DATABASES_V_DB_ROWS_DESCRIPTION)
-@RespectsInput(
-    order = 450,
-    arg = ConnectorArguments.OPT_TERADATA_MAX_DISKSPACEV_ROWS,
-    description = ConnectorArguments.TERADATA_MAX_DISK_SPACE_V_ROWS_DESCRIPTION)
 public class TeradataMetadataConnector extends AbstractTeradataConnector
     implements MetadataConnector, TeradataMetadataDumpFormat {
 
-  @SuppressWarnings("UnusedVariable")
-  private static final Logger LOG = LoggerFactory.getLogger(TeradataMetadataConnector.class);
+  public enum TeradataMetadataConnectorProperties implements ConnectorProperty {
+    TABLE_SIZE_V_MAX_ROWS(
+        "TableSizeV.max-rows", "Max number of rows to extract from dbc.TableSizeV table."),
+    DISK_SPACE_V_MAX_ROWS(
+        "DiskSpaceV.max-rows", "Max number of rows to extract from dbc.DiskSpaceV table."),
+    DATABASES_V_USERS_MAX_ROWS(
+        "DatabasesV.users.max-rows",
+        "Max number of user rows (rows with DBKind='U') to extract from dbc.DatabasesV table."),
+    DATABASES_V_DBS_MAX_ROWS(
+        "DatabasesV.dbs.max-rows",
+        "Max number of database rows (rows with DBKind='D') to extract from dbc.DatabasesV table.");
```

Comment:
The underlying table is mixed-case. Do we want these also to be mixed-case? Or is it easier for users if they're lowercase?

Reply:
You're right. It's probably easier to keep the options lowercase, while keeping the original case in the descriptions. Changed.